app/services/page_archiver.rb

```diff
@@ -0,0 +1,23 @@
+class PageArchiver
+  include Rails.application.routes.url_helpers
+
+  def self.archive!(page)
+    new(page).archive!
+  end
+
+  def self.unarchive!(page)
+    new(page).unarchive!
+  end
+
+  def initialize(page)
+    @page = page
+  end
+
+  def archive!
+    @page.update! publish_status: :archived
```

Comment:
ah yeah that makes sense I guess as it stands now `PageArchiver.archive!(page)` is just the same as `page.archive!`